aws/resource_aws_datasync_task.go

```diff
@@ -85,6 +85,16 @@ func resourceAwsDataSyncTask() *schema.Resource {
 								datasync.GidNone,
 							}, false),
 						},
+						"log_level": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  datasync.LogLevelOff,
+							ValidateFunc: validation.StringInSlice([]string{
+								datasync.LogLevelOff,
+								datasync.LogLevelBasic,
+								datasync.LogLevelTransfer,
+							}, false),
```

Comment:
The AWS SDK Go semi-recently added functions to fetch all enumeration values, e.g.

```suggestion
							ValidateFunc: validation.StringInSlice(datasync.LogLevel_Values(), false),
```